qiskit/circuit/library/standard_gates/x.py

```diff
@@ -548,28 +565,50 @@ def control(self, num_ctrl_qubits=1, label=None, ctrl_state=None):
 
     def inverse(self):
         """Invert this gate. The C3X is its own inverse."""
-        return C3XGate(angle=self._angle)
-
-    # This matrix is only correct if the angle is pi/4
-    # def to_matrix(self):
-    #     """Return a numpy.array for the C3X gate."""
-    #     return numpy.array([[1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-    #                         [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-    #                         [0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-    #                         [0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-    #                         [0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-    #                         [0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-    #                         [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-    #                         [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-    #                         [0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
-    #                         [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0],
-    #                         [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0],
-    #                         [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0],
-    #                         [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0],
-    #                         [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0],
-    #                         [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0],
-    #                         [0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=complex)
+        return C3RXGate(angle=self._angle)
+
+class C3XGate(C3RXGate):
+    def __init__(self, angle=None, label=None, ctrl_state=None):
+        import warnings
+        if angle == numpy.pi / 8:
+            warnings.warn('The parameter angle in C3XGate is going to be removed in the future. '
+                          'When angle=pi/8, use C3SqrtXGate()',
+                          DeprecationWarning, stacklevel=2)
+        elif angle == numpy.pi / 4:
+            warnings.warn('The parameter angle in C3XGate is going to be removed in the future. '
+                          'angle=pi/8, while be the default behavior C3XGate.',
+                          DeprecationWarning, stacklevel=2)
+        elif angle is not None:
+            warnings.warn('The parameter angle in C3XGate is going to be removed in the future. '
+                          'When angle!=pi/8, use C3RXGate()',
```

Comment:
```suggestion
                          'When angle!=pi/8 or pi/4, use C3RXGate()',
```

Reply:
I fixed it in 844228edddaa8f5b127994e60f45ccf3b2fcab56